packages/apollo-engine-reporting/src/extension.ts

```diff
@@ -72,9 +72,12 @@ export class EngineReportingExtension<TContext = any>
     >;
   }): EndHandler {
     // If we don't want to report a query we can just return with a empty handler
+    // We set `captureTraces` because that information is used to
     if (!this.shouldReportQuery(o.requestContext)) {
+      o.requestContext.metrics.captureTraces = false;
```

Comment:
I notice this change came with no additional tests or changes to tests - what's the effect? Is it possible for us to write tests for the effects of when this is `true` vs. `false`?

Reply:
I added tests checking if the value is set. The goal of setting the value is to have the gateway send or not send the `apollo-federation-include-trace` header to downstream services.